data/az_pwsh_cloudinit.cfg

```diff
@@ -0,0 +1,141 @@
+# To set it back set $DebugPreference = SilentlyContinue
+
+password: $pass
+chpasswd: { expire: False }
+ssh_pwauth: True
+
+runcmd:
+    # Register the Microsoft RedHat repository
+  - [ curl, -o,  "/etc/yum.repos.d/microsoft.repo", "https://packages.microsoft.com/config/rhel/7/
+  prod.repo"]
+  - [ yum,  install,  -y, powershell ]
+  - [ pwsh, -c, "Install-Module -Name Az -AllowClobber -Force" ]
+  - [ mkdir, /mnt/space ]
+
+write_files:
+  - content: |
+        # https://docs.microsoft.com/en-us/powershell/azure/authenticate-azureps
+        $tenantId = "<tenant_id>"
+        $passwd = ConvertTo-SecureString <password> -AsPlainText -Force
+        $pscredential = New-Object System.Management.Automation.PSCredential('<client_id>', $passwd)
+        Connect-AzAccount -ServicePrincipal -Credential $pscredential -TenantId $tenantId
+    path: /mnt/space/connect_account.ps1
+  - content: |
+        $PublicIPName = "<ip_name>"
+        $ResourceGroupName = "<rg_name>"
+        Get-AzPublicIpAddress -Name $PublicIPName -ResourceGroupName $ResourceGroupName
+    path: /mnt/space/get_ip.ps1
+  - content: |
+        $BlobNameSource = "<name_of_the_image, updated in the test automatically>"
+
+        # Select your subscription
+        $SubscriptionName = "<subscription_name>"
+
+        Select-AzSubscription -SubscriptionName $SubscriptionName
+
+        $ResourceGroupName = "<rg_name>"
+        $StorageAccountName = "<sa_name>"
+        $BlobSourceContainer = "<bs_container>"
+
+        # path to the image locally
+        $LocalImagePath = "/mnt/space/$BlobNameSource"
+
+        Add-AzVhd -ResourceGroupName $ResourceGroupName -Destination "http://$StorageAccountName.blob.core.windows.net/$BlobSourceContainer/$BlobNameSource" -LocalFilePath $LocalImagePath -NumberOfUploaderThreads 8 -OverWrite
+
+    path: /mnt/space/upload_vhd.ps1
+  - content: |
+        $BlobNameSource = "<name_of_the_image, updated in the test automatically>"
+        $BlobNameDest = "updated in the test automatically"
+        $VMName = "<name_of_the_vm, updated in the test automatically>"
+
+        $ResourceGroupName = "<rg_name>"
+        $StorageAccountName = "<sa_name>"
+        $BlobSourceContainer = "<bs_container>"
+
+        ## Customize for Your Environment
+
+        $BlobDestinationContainer = "<bd_container>"
+        $DeploySize= "<deploy_size>"
+
+        $InterfaceName = "<interface_name>"
+        $VNetName = "<vnet_name>"
+        $PublicIPName = "<public_ip_name>"
+
+        $StorageAccount = Get-AzStorageAccount -ResourceGroup $ResourceGroupName -Name $StorageAccountName
+
+        $SourceImageUri = "https://$StorageAccountName.blob.core.windows.net/$BlobSourceContainer/$BlobNameSource"
+        $Location = $StorageAccount.Location
+        $OSDiskName = $VMName
+
+        # Network
+        $Subnet1Name = "default"
+        $VNetAddressPrefix = "10.1.0.0/16"
+        $VNetSubnetAddressPrefix = "10.1.0.0/24"
```

Comment:
I think we can put this with Azure in data yaml?

Reply:
@digitronik it's not a secret value https://docs.microsoft.com/en-us/azure/virtual-network/virtual-networks-faq

```
What address ranges can I use in my VNets?
Any IP address range defined in RFC 1918. For example, 10.0.0.0/16. 
```